### ISSUE TITLE
Collect Jabber interfaces in one file

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2177]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#2177|
|Type|task|
|Created|2006-10-19 14:53:58Z|

Interface declarations in the jabber module are currently in the same files as their implementation classes. Create a new `t.w.p.j.ijabber` module and move all interfaces there.

<details><summary>Searchable metadata</summary>

```
trac-id__2177 2177
type__task task
reporter__ralphm ralphm
priority__highest highest
milestone__Words_0_5 Words-0.5
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1161269638000000 1161269638000000
changetime__1164622471000000 1164622471000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_3"></a>|@ralphm removed owner|
|-|-|

Implemented in [branches/jabber-interfaces-2177.](branches/jabber-interfaces-2177.) Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set owner to @ralphm|
|-|-|

I can't switch to this branch using chbranch, and when I merged manually (`svn merge -r18558:18559 $TREPO/branches/jabber-interfaces-2177 .`), I got a conflict in jabber/xmlstream.py.

Can you please merge forward, resolving conflicts, and re-assign to me for review?

Sorry for the delay in getting around to this.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_6"></a>|@ralphm set owner to @jml|
|-|-|

Odd, I had no problems with chbranch. I did get the same conflict, now resolved in jabber-interfaces-2177-2. Please review again.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set owner to @ralphm|
|-|-|

Looks good. I would have just done from t.w.ijabber import * (one of the strengths of putting all the interfaces into a single module), but what you've done is fine, perhaps better.

Merge away!

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_8"></a>|@ralphm set status to `closed`|
|-|-|

(In [18775]) Move public Jabber interface definitions to one module.

Author: ralphm
Reviewer: jml
Fixes [#2177](https://github.com/twisted/twisted/issues/2177)

The interface definitions in sasl_mechanisms are left there, because this
module should really move to or be merged with a more generic SASL
implementation (see [#2015](https://github.com/twisted/twisted/issues/2015)).

The interface definitions in xmpp_stringprep are left there, because they
are considered private and for documentation only.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
